python/pyspark/sql/pandas/functions.pyi

```diff
@@ -46,13 +46,21 @@ class PandasUDFType:
     GROUPED_MAP: PandasGroupedMapUDFType
     GROUPED_AGG: PandasGroupedAggUDFType
 
+@overload
+def pandas_udf(
+    f: PandasScalarToScalarFunction, returnType: Union[AtomicDataTypeOrString, ArrayType]
+) -> UserDefinedFunctionLike: ...
 @overload
 def pandas_udf(
     f: PandasScalarToScalarFunction,
     returnType: Union[AtomicDataTypeOrString, ArrayType],
     functionType: PandasScalarUDFType,
 ) -> UserDefinedFunctionLike: ...
 @overload
+def pandas_udf(
+    f: Union[AtomicDataTypeOrString, ArrayType]
```

Comment:
While we're here, let's just add all data type instead of only Atomic type.

Reply:
Add some other overloads which are same to these two to cover all situations.  (with some ignores to pass `mypy` checks like those existing ignores)